mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+                        crystal_acronymes[0] if len(crystal_acronymes) > 0 else ""
+                    )
+                lst = {
+                    "crystal_uuid": x_tal,
+                    "name": crystal_names[index],
+                    "state": sample_states[index],
+                    "acronym": proteinAcronym,
+                    "img_url": img_url,
+                    "img_target_x": img_target_x,
+                    "img_target_y": img_target_y,
+                }
+                crystal_list.append(lst)
+        except Exception as ex:
+            print("Could not get Crystal List : %s" % str(ex))
+            # TEMP return a fake list for test
+            crystal_list = [
+                {
+                    "crystal_uuid": "c9ca5e0d-35fd-4ff9-9ad2-4de9fd47fd83",
+                    "name": "TEST1",
+                    "state": "ready_to_execute",
+                    "acronym": "cryoprotectant",
+                    "img_url": "https://htxlab.embl.org//rawimages/2021//CD032401/6/FORMULATRIX_CD032401_6_08-09-2021_05_02_01_00_99_Vis.jpg",
+                    "img_target_x": 26.868617890692,
+                    "img_target_y": 50.122652377553,
+                },
+                {
+                    "crystal_uuid": "94730c39-bf66-416f-ab97-f755e45f6a3a",
+                    "name": "TEST12",
+                    "state": "needs_repositionning",
+                    "acronym": "cryoprotectant",
+                    "img_url": "https://htxlab.embl.org//rawimages/2021//CD032401/6/FORMULATRIX_CD032401_6_08-09-2021_06_04_01_00_99_Vis.jpg",
+                    "img_target_x": 54.466230936819,
+                    "img_target_y": 58.026175213675,
+                },
+                {
+                    "crystal_uuid": "5d5dfc88-009d-4486-8526-33bf58c4d5d9",
+                    "name": "TEST13",
+                    "state": "failed",
+                    "acronym": "cryoprotectant",
+                    "img_url": "https://htxlab.embl.org//rawimages/2021//CD032401/6/FORMULATRIX_CD032401_6_08-09-2021_06_02_01_00_99_Vis.jpg",
+                    "img_target_x": 32.710943173566,
+                    "img_target_y": 51.110576923077,
+                },
+                {
+                    "crystal_uuid": "46aca4a4-0c00-4685-b1e1-53071bdcb66d",
+                    "name": "TEST4",
+                    "state": "harvested",
+                    "acronym": "cryoprotectant",
+                    "img_url": "https://htxlab.embl.org//rawimages/2021//CD032401/6/FORMULATRIX_CD032401_6_08-09-2021_06_04_01_00_99_Vis.jpg",
+                    "img_target_x": 38.632400075415,
+                    "img_target_y": 41.762286324786,
+                },
+            ]
+
+        return crystal_list
+
+    def get_global_state(self):
+        try:
+            return HWR.beamline.harvester_maintenance.get_global_state()
+        except Exception:
+            return "OFFLINE", "OFFLINE", "OFFLINE"
```

Comment:
I really do not like those catch-all exceptions. There might be some actual issues somewhere in this block of code and we will never know about it, because it is swallowed and there is no logging! Is there really no cleaner way to handle this?